02-csv-vector-raster-plotting.Rmd

```diff
@@ -78,271 +92,407 @@ Converting from .csv to Shapefile in R</a>
- * What is the coordinate reference system (CRS) for this data? 
- 
+#Determine a Coordinate Reference System
+When we create a spatial points `data.frame` from a .csv, there is no inherent
+CRS associated with it. We must assign a CRS based knowledge of the coordinate
+system the x,y values are in.
+
+How do we know what the correct CRS for the data is? There are several ways we 
+know this.  
+
+1.  We collected the data ourselves and know from the GPS what the CRS, x,y
+values, and units are. 
+2. The source of our .csv file for the data points also has a .txt (or similar)
+file with metadata about the data.  For more information on metadata in
+associated files check out the
+[Understanding Time Series Metadata]({{site.baseurl}}/R/Time-Series-Metadata) 
+lesson. 
+3. We look at the data in our .csv (now `R` object) to see if it gives any
+hints.
+
+```{r crs-sleuthing}
+#view the column names
+names(plot.locations_HARV)
+```
+
+Okay, there are several colums related to a CRS.  Let's check out the first to
+rows of the `data.frame` to see what information is in these columns.
+
+```{r crs-sleuthing2}
+#view the first 6 rows
+head(plot.locations_HARV)
+```
+
+We have the following information that is relevant to a CRS: 
+* X: 73xxxx.3 -- no other information but these match the eastings
+* Y: 4713xxx -- no other information but these match the northings
+* decimalLat: 42.5xxxx
+* decimalLon: -72.1xxx
+* geodeticDa: WGS84  -- this is geodetic datum WGS84
+* utmZone: 18N
+* easting: 73xxxx.3
+* northing: 4713xxx
+
+What can we make our of this?   We have a geographic coordinate system (Latitude
+and Longitude, both in decimal degrees) and a projected coordinate system (UTM
+in Zone18N with Eastings and Northings).  Importantly for the projection we know
+that the datum is WGS84.  
+
+#Convert .csv to Shapefile
+Before we can combine our plot locations with the boundaries of our field site, 
+or other spatial data we need to convert the locations from a .csv file to a
+`SpatialPointsDataFrame`.  When we make the conversion we also have to specify 
+what the correct `crs`, datum, and other information.  We do this using a
+`proj4string`.  As we are going to be projecting this data on a 2D surface to
+make a map. We'll convert to a SpatialPointsDataFrame based on the UTM
+coordinates. 
+
+What is a `proj4string`? This is a string with specific information about the
+`crs` and related information for the data in a spatial file.  The format
+changes depending on the projection used.  This format is for UTMs. 
+`+proj=utm +zone=18 +datum=WGS84 +units=m +no_defs +ellps=WGS84 +towgs84=0,0,0`
+
+* +proj=utm     the projection -- we know our data is in UTM
+* +zone=18      the UTM zone -- we know our data is in 18N but N isnt' needed
+* +datum=WGS84  the datum -- we know our data is in WGS84
+* +units=m      the coordinate value units -- UTM is by default meters (m)
+* +no_defs      with rare exceptions this default used 
```

Comment:
(okay this is my BSing based on what I could find. I couldn't figure out what this actual stood for)